### PR TITLE
Work around incorrect GCC warning

### PR DESCRIPTION
**Pull request by nbougalis**
Testing.

### CODE REVIEW

File: include/beast/zlib/detail/deflate_stream.hpp [hunk 1]
```diff
@@ -1060,7 +1060,7 @@ doWrite(z_params& zs, boost::optional<Flush> flush, error_code& ec)
         }
     }
     else if(zs.avail_in == 0 && (
-            old_flush && flush <= *old_flush
+            old_flush && flush <= old_flush
```
Comment (vinniefalco) [inline include/beast/zlib/detail/deflate_stream.hpp]: Yeah that's how it was before...

**Comment by vinniefalco** [comment]
bzzzzzt wrong, thanks for playing, please try again ;) ;) ;)
https://travis-ci.org/vinniefalco/Beast/jobs/252620909#L6266

**Comment by uecasm** [comment]
Shouldn't that code instead be doing a `swap`?  Or at least using move-assignment.

**Comment by vinniefalco** [comment]
**swap** ?! this is an enum

**Comment by vinniefalco** [comment]
Merged!

**Pull request closed without merging**